rotkehlchen/db/search_assets.py

```diff
@@ -0,0 +1,119 @@
+    levenshtein_distances: List[int] = []
+    query, bindings = GlobalDBHandler()._prepare_search_assets_query(filter_query)
+    resolved_eth = A_ETH.resolve_to_crypto_asset()
+    with GlobalDBHandler().conn.read_ctx() as globaldb_cursor, db.conn.read_ctx() as cursor:
+        globaldb_cursor.execute(query, bindings)
+        treat_eth2_as_eth = db.get_settings(cursor).treat_eth2_as_eth
+        found_eth = False
+        for entry in globaldb_cursor:
+            # if an asset does not have a name and symbol, skip.
+            if entry[1] is None and entry[2] is None:
+                continue
+
+            substr_in_name = substr_in_symbol = None
+            # default to a value greater than the threshold to avoid false-positives
+            # and prevent raising `UnboundLocalError` when name/symbol is None.
+            lev_dist_name = lev_dist_symbol = 6
+            if entry[1] is not None:
+                name_casefold = entry[1].casefold()
+                lev_dist_name = levenshtein(substring_search, name_casefold)
+                substr_in_name = substring_search in name_casefold
+
+            if entry[2] is not None:
+                symbol_casefold = entry[2].casefold()
+                lev_dist_symbol = levenshtein(substring_search, symbol_casefold)
+                substr_in_symbol = substring_search in symbol_casefold
+
+            if not substr_in_name and not substr_in_symbol:
+                continue
+
+            lev_dist_min = min(lev_dist_name, lev_dist_symbol)
+            if treat_eth2_as_eth is True and entry[0] in (A_ETH.identifier, A_ETH2.identifier):  # noqa:E501
+                if found_eth is False:
+                    search_result.append({
+                        'identifier': resolved_eth.identifier,
+                        'name': resolved_eth.name,
+                        'symbol': resolved_eth.symbol,
+                        'asset_type': 'own chain',
+                    })
+                    levenshtein_distances.append(lev_dist_min)
+                    found_eth = True
+                continue
+
+            entry_info = {
+                'identifier': entry[0],
+                'name': entry[1],
+                'symbol': entry[2],
+                'asset_type': AssetType.deserialize_from_db(entry[4]).serialize(),
+            }
+            if entry[3] is not None:
+                entry_info['evm_chain'] = ChainID.deserialize_from_db(entry[3]).serialize()
+            if entry[5] is not None:
+                entry_info['custom_asset_type'] = entry[5]
+
+            search_result.append(entry_info)
+            levenshtein_distances.append(lev_dist_min)
+
+        if search_nfts is True:
+            cursor.execute('SELECT identifier, name, collection_name FROM nfts')
+            for entry in cursor:
+                # if an nft does not have a name and collectio name, skip.
+                if entry[1] is None and entry[2] is None:
+                    continue
```

Comment:
Can't this logic be moved to its own function? You are not using anything from the asset logic if I understood correctly

Reply:
maybe, yeah

Reply:
Is not abstract it. Just move it to a an auxiliar method `_search_nfts_levenshtein`. This function grow a lot and is hard to follow it as it is now. Since the nft logic doesn't depend on the rest of the logic you could move it